UnityProject/Assets/Scripts/Core/Lifecycle/CleanupUtil.cs

```diff
@@ -0,0 +1,194 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEditor;
+using System;
+using UI.Action;
+using Audio.Containers;
+
+public static class CleanupUtil
+{
+
+	public static int RidListOfSoonToBeDeadElements<T>(IList<T> list_in_question, Func<T, UnityEngine.MonoBehaviour> target_extractor)
```

Comment:
could you add an option, to make it complain and log everything  so people can work out what leaking if debugging  ?

Reply:
Done in c5c93cf411cf9888714a492bc12d7820d9f0f465